cdap-ui/app/features/admin/controllers/namespace/templates-list-ctrl.js

```diff
@@ -1,27 +1,53 @@
 angular.module(PKG.name + '.feature.admin')
-  .controller('NamespaceTemplatesListController', function ($scope, mySettings, $stateParams, $alert) {
+  .controller('NamespaceTemplatesListController', function ($scope, mySettings, $stateParams, $alert, myHelpers) {
 
     var vm = this;
-    vm.list = {};
+    vm.list = [];
+
+    function objectToArray(obj) {
+      angular.forEach(obj, function (value) {
+        vm.list.push(value);
+      });
+    }
+
+    function processResult(response) {
+      if (response) {
+        vm.list = [];
+
+        var sources, transforms, sinks;
+
+        sources = myHelpers.objectQuery(response, $stateParams.nsadmin, 'ETLBatch', 'source');
```

Comment:
This could wrapped something like,

```Javascript

function processResult(response) {

  if (!response) {
     return;
  } 
  vm.list = [];
  ['ETLBatch', 'ETLRealtime'].forEach(function(templatetype) {
     ['source', 'transforms', 'sink'].forEach(function(plugintype) {
       var pluginArray = objectToArray(myHelpers.objectQuery(response, $stateParams.nsadmin, templatetype, plugintype));
       vm.list = vm.list.concat(pluginArray);
     });
  });
}
function objectToArray(obj) {
  var returnArray = [];
   angular.forEach(obj, function (value) {
     returnArray.push(value);
  });
  return returnArray;
}
```

This way we know where and who updates ```vm.list``` and there is no repetition of code. Right now vm.list is manipulated in three different places and by looking at processResult I don't see why we use that function until I see ```objectToArray``` function.